### PR TITLE
Release 6.0.0

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Fixes from release testing.

## Changes

* Replaced mistaken key in request_status lambdas with correct key
* Fixed schema imports
* Fixed bad key in copy_to_glacier logging statement

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## How Has This Been Tested?

Deployed and run in AWS.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [x] Development documentation
    - [x] User documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: tasks/copy_to_glacier/copy_to_glacier.py [hunk 1]
```diff
@@ -223,10 +223,8 @@ def task(event: Dict[str, Union[List[str], Dict]], context: object) -> Dict[str,
             file_hash_type = file.get(FILE_HASH_TYPE_KEY, None)
             if should_exclude_files_type(file_filepath, exclude_file_types):
                 LOGGER.info(
-                    "Excluding {file_filepath} from glacier backup "
-                    "because of collection configured {CONFIG_EXCLUDED_FILE_EXTENSIONS_KEY}.",
-                    file_filepath=file_filepath,
-                    CONFIG_ORCA_EXCLUDED_FILE_EXTENSIONS_KEY=CONFIG_EXCLUDED_FILE_EXTENSIONS_KEY,
+                    f"Excluding {file_filepath} from glacier backup "
+                    f"because of collection configured {CONFIG_EXCLUDED_FILE_EXTENSIONS_KEY}."
```
Comment (AndrewDorn) [inline tasks/copy_to_glacier/copy_to_glacier.py:227]: These keys were incorrect, causing KeyErrors

File: tasks/post_copy_request_to_queue/bin/build.sh [hunk 2]
```diff
@@ -108,6 +108,13 @@ let return_code=$?
 
 check_rc $return_code "ERROR: Failed to copy lambda files to build directory."
 
+## Copy the schema files to build
+echo "INFO: Copying schema files ..."
+cp -r schemas/ build/
+let return_code=$?
+
+check_rc $return_code "ERROR: Failed to copy schema files to build directory."
+
```
Comment (AndrewDorn) [inline tasks/post_copy_request_to_queue/bin/build.sh:117]: I'm guessing this was using a non-standard method of pulling in schema files, and I messed it up when I removed the redundant copies.

File: tasks/request_status_for_granule/API.md [hunk 3]
```diff
@@ -164,7 +159,7 @@ Entry point for the request_status_for_granule Lambda.
 - `'restore_destination'` _str_ - The name of the glacier bucket
   the file is being copied to.
 - `'status'` _str_ - The status of the restoration of the file.
-  May be 'pending', 'staged', 'success', or 'failed'.
+  May be 'pending', 'staged', 'success', or 'error'.
```
Comment (AndrewDorn) [inline tasks/request_status_for_granule/API.md:162]: Fixed this in a lot of places. DB table uses the word "error", and that's what we key off of.

File: tasks/request_status_for_granule/request_status_for_granule.py [hunk 4]
```diff
@@ -55,21 +63,11 @@ def task(
...
 
     job_entry = get_job_entry_for_granule(granule_id, job_id, engine)
     if job_entry is None:
-        return create_http_error_dict(
-            "NotFound",
-            HTTPStatus.NOT_FOUND,
-            request_id,
-            f"No job found for granule id '{granule_id}' and job id '{job_id}'.",
-        )
+        raise JobGranuleCombinationNotFoundException(job_id, granule_id)
```
Comment (AndrewDorn) [inline tasks/request_status_for_granule/request_status_for_granule.py:70]: Pulled AWS specific error-handling out to handler function here and in request_status_for_job.
As-is, handler didn't know if it should check against the output.json or if an error occurred.

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,19 +13,24 @@ and includes an additional section for migration notes.
...
   Both lambdas will return proper HTTP error codes for bad inputs of internal server errors.
+  Additionally, corrected error in [API Reference](https://nasa.github.io/cumulus-orca/docs/developer/api/orca-api) 
+  where the `error` status for these lambdas was incorrectly listed as `failed`.
+- *ORCA-320* Requests to API Gateway now use IAM permissions, restricting anonymous access.
 
 ### Migration Notes
 
-- Adjust workflows/step functions for `OrcaCopyToGlacierWorkflow`.
+- Adjust workflow/step-function usage for new key paths.
+  - Affects `OrcaCopyToGlacierWorkflow` and `OrcaRecoveryWorkflow`.
```
Comment (bhazuka) [inline CHANGELOG.md:33]: I always think of migration notes as something that affects external users using the software. In this case, `OrcaCopyToGlacierWorkflow` and `OrcaRecoveryWorkflow` is something the ORCA team owns and manages. The changes really affect any step functions that the end user has that calls the `copy_to_glacier` task. I am wondering if we should reword this to be something like:
```
Adjust workflow/step-function usage for items that use `copy_to_glavier` task for new key paths.
```
Thoughts?

File: CHANGELOG.md [hunk 6]
```diff
@@ -13,19 +13,24 @@ and includes an additional section for migration notes.
...
 - *ORCA-290* Adjusted workflows/step functions for `OrcaRecoveryWorkflow`.
   - `excludeFileTypes`, `orcaDefaultBucketOverride` and `orcaDefaultStorageClassOverride` arguments in `task_config` are now `excludedFileExtensions`, `defaultBucketOverride` and  `defaultStorageClassOverride` respectively.
   - `excludedFileExtensions`, `defaultBucketOverride` and `defaultStorageClassOverride` keys are now under `collection.meta.orca`. See the example below under `Migration Notes`.
 - *ORCA-519* Enforced schema checks in `request_status_for_granule` and `request_status_for_job`.
   Both lambdas will return proper HTTP error codes for bad inputs of internal server errors.
+  Additionally, corrected error in [API Reference](https://nasa.github.io/cumulus-orca/docs/developer/api/orca-api) 
+  where the `error` status for these lambdas was incorrectly listed as `failed`.
+- *ORCA-320* Requests to API Gateway now use IAM permissions, restricting anonymous access.
```
Comment (bhazuka) [inline CHANGELOG.md:28]: Should also probably add [ORCA-496](https://bugs.earthdata.nasa.gov/browse/ORCA-496) Mitigated SQS security issue. All SQS queues now use default encryption.

File: tasks/copy_to_glacier/copy_to_glacier.py [hunk 7]
```diff
@@ -223,10 +223,8 @@ def task(event: Dict[str, Union[List[str], Dict]], context: object) -> Dict[str,
             file_hash_type = file.get(FILE_HASH_TYPE_KEY, None)
             if should_exclude_files_type(file_filepath, exclude_file_types):
                 LOGGER.info(
-                    "Excluding {file_filepath} from glacier backup "
-                    "because of collection configured {CONFIG_EXCLUDED_FILE_EXTENSIONS_KEY}.",
-                    file_filepath=file_filepath,
-                    CONFIG_ORCA_EXCLUDED_FILE_EXTENSIONS_KEY=CONFIG_EXCLUDED_FILE_EXTENSIONS_KEY,
+                    f"Excluding {file_filepath} from glacier backup "
```
Comment (bhazuka) [inline tasks/copy_to_glacier/copy_to_glacier.py:226]: Should we change `glacier` here to `ORCA archive`?

File: tasks/copy_to_glacier/copy_to_glacier.py [hunk 8]
```diff
@@ -245,13 +243,11 @@ def task(event: Dict[str, Union[List[str], Dict]], context: object) -> Dict[str,
             result["hashType"] = file_hash_type
             copied_file_urls.append(file_source_uri)
             LOGGER.info(
-                "Copied {file_filepath} into glacier storage bucket {default_bucket}.",
-                file_filepath=file_filepath,
-                default_bucket=destination_bucket,
+                f"Copied {file_filepath} into glacier storage bucket {destination_bucket}."
```
Comment (bhazuka) [inline tasks/copy_to_glacier/copy_to_glacier.py:246]: Should we change `glacier` here to `ORCA archive`?

File: tasks/copy_to_glacier/copy_to_glacier.py [hunk 9]
```diff
@@ -245,13 +243,11 @@ def task(event: Dict[str, Union[List[str], Dict]], context: object) -> Dict[str,
...
             )
             # Add file record to metadata SQS message
             LOGGER.debug(
-                "Adding the files dictionary to the SQS body {result}.", result=result
+                "Adding the files dictionary to the SQS body. {result}.", result=result
```
Comment (bhazuka) [inline tasks/copy_to_glacier/copy_to_glacier.py:250]: Should this be changed to `f` string like the others you fixed?

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 10]
```diff
@@ -84,11 +82,13 @@ the release process is complete.
...
 
-Finally, you need to merge the version update changes back into develop and 
-master.
+If this is the latest version, you need to merge the version update changes back into develop, then synchronize develop and master.
 
-If this is the latest version, create the PRs to merge the release branch 
-into develop and master. 
+1. Create a PR to merge the release branch into develop.
+   This should only contain changes from the release process.
+1. Once complete, create a PR to merge develop into master.
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/versioning-releases.md:89]: This could be problematic and I would not recommend merging develop into master. Unless you plan on freezing any PR merges into develop when you are working on a release. This could cause untested and unplanned changes being released into master.

To avoid this, I would only merge the release branch into master.

**Review by bhazuka — CHANGES_REQUESTED** [review]
Couple of thoughts and ideas. Overall looks good though.

File: CHANGELOG.md (hunk 5 above)
Comment (AndrewDorn) [reply CHANGELOG.md:33]: Added separate line for CopyToGlacier usage, as I recall that some customers include copy_to_glacier in their own workflows.
Leaving the other section for now, as anyone integrating with our workflows still needs to know about the key changes.

File: CHANGELOG.md (hunk 6 above)
Comment (AndrewDorn) [reply CHANGELOG.md:28]: Added

File: tasks/copy_to_glacier/copy_to_glacier.py (hunk 9 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/copy_to_glacier.py:250]: Nope. Result can contain `{}` and Cumulus logger treats that as requiring a parameter for replacement.

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 10 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/versioning-releases.md:89]: True. Updated.

File: tasks/copy_to_glacier/copy_to_glacier.py (hunk 9 above)
Comment (bhazuka) [reply tasks/copy_to_glacier/copy_to_glacier.py:250]: ahh, ok.

File: tasks/copy_to_glacier/copy_to_glacier.py (hunk 7 above)
Comment (bhazuka) [reply tasks/copy_to_glacier/copy_to_glacier.py:226]: ORCA-532 created for this work.

File: tasks/copy_to_glacier/copy_to_glacier.py (hunk 8 above)
Comment (bhazuka) [reply tasks/copy_to_glacier/copy_to_glacier.py:246]: ORCA-532 created for this work.

**Review by bhazuka — APPROVED** [review]
